### PR TITLE
Add test style guide

### PR DESCRIPTION
**Pull request by zekth**
Improving style guide for tests as mentionned in : https://github.com/denoland/deno_std/issues/130

### CODE REVIEW

File: website/style_guide.md [hunk 1]
```diff
@@ -262,3 +262,39 @@ the first column of the comment. For example:
...
+## Each module should come with tests
+
+Each module should come with its test as a sibling with the name
+`modulename_test.ts`. For example the module `foo.ts` should come with its
+sibling `foo_test.ts`.
+
+## Unit Tests should be explicit
+
+For a better understanding of the tests, function should be correctly named as
+its prompted throught the test command. Like:
```
Comment (ry) [inline website/style_guide.md]: s/throught/throughout/

File: website/style_guide.md [hunk 2]
```diff
@@ -262,3 +262,39 @@ the first column of the comment. For example:
...
+
+```
+test myTestFunction ... ok
+```
+
+Example of test:
+
+```ts
+import { assertStrContains } from "./asserts.ts";
+import { test } from "./mod.ts";
```
Comment (ry) [inline website/style_guide.md]: Use the absolute url https://deno.land/std@v0.3.1/testing/mod.ts

File: website/style_guide.md [hunk 3]
```diff
@@ -262,3 +262,39 @@ the first column of the comment. For example:
...
+test myTestFunction ... ok
+```
+
+Example of test:
+
+```ts
+import { assertStrContains } from "https://deno.land/std@v0.3.1/testing/asserts.ts";
+import { test } from "https://deno.land/std@v0.3.1/testing/mod.ts";
+
+test(function testingAssertStringContains() {
```
Comment (ry) [inline website/style_guide.md]: Maybe the name of the function here should correspond to the output you give above?

Comment (zekth) [reply website/style_guide.md]: Didn't thought about it but yes

File: website/style_guide.md [hunk 4]
```diff
@@ -262,3 +262,39 @@ the first column of the comment. For example:
...
+  assertStrContains("Denosaurus", "Deno");
+  assertStrContains("Denosaurus", "rus");
+  let didThrow;
+  try {
+    assertStrContains("Denosaurus", "Raptor");
+    didThrow = false;
+  } catch (e) {
+    didThrow = true;
+  }
+  assertEquals(didThrow, true);
```
Comment (ry) [inline website/style_guide.md]: Please use a smaller one line example with just a single `assertEquals`... I think all this code distracts from the point the style guide is making. This isn't documentation.

Comment (zekth) [reply website/style_guide.md]: edited

**Pull request merged**